app/src/main/java/org/jellyfin/androidtv/ui/itemhandling/ItemLauncher.java

```diff
@@ -139,17 +138,28 @@ public static void launch(final BaseRowItem rowItem, ItemRowAdapter adapter, int
 
                     case Audio:
                         Timber.d("got pos %s", pos);
-                        // if a song isn't the first in the queue, play it
-                        if (rowItem.getIndex() > KoinJavaComponent.<MediaManager>get(MediaManager.class).getCurrentAudioQueuePosition() && rowItem.getIndex() < KoinJavaComponent.<MediaManager>get(MediaManager.class).getCurrentAudioQueueSize()) {
-                            KoinJavaComponent.<MediaManager>get(MediaManager.class).playFrom(rowItem.getIndex());
-                        } else {
+                        if (rowItem.getBaseItem() == null)
+                            return;
+                        // if the song currently playing is selected (and is the exact item - this only happens in the nowPlayingRow), open AudioNowPlayingActivity
+                        if (KoinJavaComponent.<MediaManager>get(MediaManager.class).hasAudioQueueItems() && rowItem.getBaseItem() == KoinJavaComponent.<MediaManager>get(MediaManager.class).getCurrentAudioItem()) {
                             // otherwise, open AudioNowPlayingActivity
                             Intent nowPlaying = new Intent(activity, AudioNowPlayingActivity.class);
                             activity.startActivity(nowPlaying);
+                        } else if (KoinJavaComponent.<MediaManager>get(MediaManager.class).hasAudioQueueItems() && rowItem instanceof AudioQueueItem && pos < KoinJavaComponent.<MediaManager>get(MediaManager.class).getCurrentAudioQueueSize()) {
```

Comment:
Probably more efficient (and readable) to assign the mediamanager to a variable here instead of writing `KoinJavaComponent.<MediaManager>get(MediaManager.class)` multiple times.